api/v1/dcluster_types_extra.go

```diff
@@ -0,0 +1,89 @@
+	Driver                 *dbmodels.SparkNode         `json:"driver,omitempty" url:"driver,omitempty"`
+	Executors              []dbmodels.SparkNode        `json:"executors,omitempty" url:"executors,omitempty"`
+	SparkContextID         int64                       `json:"spark_context_id,omitempty" url:"spark_context_id,omitempty"`
+	JdbcPort               int32                       `json:"jdbc_port,omitempty" url:"jdbc_port,omitempty"`
+	ClusterName            string                      `json:"cluster_name,omitempty" url:"cluster_name,omitempty"`
+	SparkVersion           string                      `json:"spark_version,omitempty" url:"spark_version,omitempty"`
+	SparkConf              *dbmodels.SparkConfPair     `json:"spark_conf,omitempty" url:"spark_conf,omitempty"`
+	NodeTypeID             string                      `json:"node_type_id,omitempty" url:"node_type_id,omitempty"`
+	DriverNodeTypeID       string                      `json:"driver_node_type_id,omitempty" url:"driver_node_type_id,omitempty"`
+	ClusterLogConf         *dbmodels.ClusterLogConf    `json:"cluster_log_conf,omitempty" url:"cluster_log_conf,omitempty"`
+	InitScripts            []dbmodels.InitScriptInfo   `json:"init_scripts,omitempty" url:"init_scripts,omitempty"`
+	SparkEnvVars           map[string]string           `json:"spark_env_vars,omitempty" url:"spark_env_vars,omitempty"`
+	AutoterminationMinutes int32                       `json:"autotermination_minutes,omitempty" url:"autotermination_minutes,omitempty"`
+	State                  *dbmodels.ClusterState      `json:"state,omitempty" url:"state,omitempty"`
+	StateMessage           string                      `json:"state_message,omitempty" url:"state_message,omitempty"`
+	StartTime              int64                       `json:"start_time,omitempty" url:"start_time,omitempty"`
+	TerminateTime          int64                       `json:"terminate_time,omitempty" url:"terminate_time,omitempty"`
+	LastStateLossTime      int64                       `json:"last_state_loss_time,omitempty" url:"last_state_loss_time,omitempty"`
+	LastActivityTime       int64                       `json:"last_activity_time,omitempty" url:"last_activity_time,omitempty"`
+	ClusterMemoryMb        int64                       `json:"cluster_memory_mb,omitempty" url:"cluster_memory_mb,omitempty"`
+	ClusterCores           string                      `json:"cluster_cores,omitempty" url:"cluster_cores,omitempty"`
+	DefaultTags            []dbmodels.ClusterTag       `json:"default_tags,omitempty" url:"default_tags,omitempty"`
+	ClusterLogStatus       *dbmodels.LogSyncStatus     `json:"cluster_log_status,omitempty" url:"cluster_log_status,omitempty"`
+	TerminationReason      *dbmodels.TerminationReason `json:"termination_reason,omitempty" url:"termination_reason,omitempty"`
+}
+
+func (i *DclusterInfo) FromDataBricksClusterInfo(ci dbmodels.ClusterInfo) *DclusterInfo {
```

Comment:
rename "i" to "dci" to have consistent naming convention

Reply:
Renamed the variable name as suggested. Also added reason why this method is needed.